client/my-sites/plans-grid/hooks/npm-ready/use-is-large-currency.ts

```diff
@@ -16,48 +59,25 @@ interface Props {
  * 9 characters. For example, $4,000 undiscounted and $30 discounted would be 9 characters.
  * This is primarily used for lowering the font-size of "large" display prices.
  */
-export default function useIsLargeCurrency( { gridPlans, returnMonthly = true }: Props ) {
-	return gridPlans.some( ( gridPlan ) => {
-		const {
-			pricing: { originalPrice, discountedPrice, currencyCode },
-		} = gridPlan;
-
-		/**
-		 * Because this hook is primarily used for lowering font-sizes of "large" display prices,
-		 * this implementation is non-ideal. It assumes that each character in the display price,
-		 * including currency symbols, comma separators, decimal points, etc. are all the same width.
-		 *
-		 * For the time being, however, this is a good enough approximation and works with our existing
-		 * currencies. We'll need to update this in the near future, especially to prevent issues with
-		 * newly introduced currencies.
-		 *
-		 * https://github.com/Automattic/wp-calypso/pull/81537#discussion_r1323182287
-		 */
-		const priceCharacterCounts = [
-			originalPrice[ returnMonthly ? 'monthly' : 'full' ],
-			discountedPrice[ returnMonthly ? 'monthly' : 'full' ],
-		].map( ( price ) => {
-			/**
-			 * Prices are represented in smallest units for a currency, and not as prices that
-			 * are actually displayed. Ex. $20 is the integer 2000, and not 20. To determine if
-			 * the display price is too long, we convert the integer to a display string.
-			 */
-			return price
-				? formatCurrency( price, currencyCode || 'USD', {
-						stripZeros: true,
-						isSmallestUnit: true,
-				  } ).length
-				: 0;
-		} );
-
-		const exceedsSingleCountThreshold = priceCharacterCounts.some(
-			( charLength ) => charLength > LARGE_CURRENCY_CHAR_THRESHOLD
-		);
-
-		const exceedsCombinedCountThreshold =
-			priceCharacterCounts.reduce( ( acc, price ) => acc + price, 0 ) >
-			LARGE_CURRENCY_COMBINED_CHAR_THRESHOLD;
-
-		return exceedsSingleCountThreshold || exceedsCombinedCountThreshold;
-	} );
+export default function useIsLargeCurrency( { prices, isAddOn = false, currencyCode }: Props ) {
```

Comment:
One thing worth thinking is whether the prices are from add-ons or plans is a concept that `useIsLargeCurrency()` should be aware of. By doing so, we actually introduce a conceptual coupling that `useIsLargeCurrency()` is aware of what's the right threshould values for certain UI components. One potential way to resolve this conceptual coupling is to let the caller decide the threshold value instead, so it becomes a general utility function.

However, it's not a big deal for now since we don't use it anywhere else, so let's keep it :)

Reply:
> we actually introduce a conceptual coupling that useIsLargeCurrency() is aware of what's the right threshould values for certain UI components

Thanks for bringing this up! This crossed my mind briefly, but it seemed like overkill for our current use cases. I can add a note in the tracking issue to keep this in consideration 🙂